src/compiler/types.ts

```diff
@@ -1582,14 +1582,15 @@ module ts {
         Tuple                   = 0x00002000,  // Tuple
         Union                   = 0x00004000,  // Union
         Anonymous               = 0x00008000,  // Anonymous
+        Instantiated            = 0x00010000,  // Instantiated anonymous type

```

Comment:
Can you rename this InstantiatedAnonymous? Or have all the types created by instantiateType take on the flag as well.

Also, it seems to me that References are always Instantiated. To that end, can you OR this flag into the Reference flag as well so the subset / superset relation is clear?